lib/schedule.module.ts

```diff
@@ -4,17 +4,38 @@ import { SchedulerMetadataAccessor } from './schedule-metadata.accessor';
 import { ScheduleExplorer } from './schedule.explorer';
 import { SchedulerOrchestrator } from './scheduler.orchestrator';
 import { SchedulerRegistry } from './scheduler.registry';
+import { ScheduleModuleOptions } from './interfaces/schedule-module-options.interface';
+import { SCHEDULE_MODULE_OPTIONS } from './schedule.constants';
 
 @Module({
   imports: [DiscoveryModule],
   providers: [SchedulerMetadataAccessor, SchedulerOrchestrator],
 })
 export class ScheduleModule {
-  static forRoot(): DynamicModule {
+  static forRoot(
+    options: ScheduleModuleOptions = {
+      cronJobs: true,
+      intervals: true,
+      timeouts: true,
+    },
```

Comment:
can make `options` optional and skip the default value in the method signature then (since you have `optionsWithDefaults` below anyway)